### PR TITLE
Adds drop effect to dropped random option items

### PR DESCRIPTION
**Pull request by aleos89**
* **Addressed Issue(s)**: N/A

* **Server Mode**: Renewal

* **Description of Pull Request**: 
  * Adds the official pillar effect on dropped items depending on how many random options were applied to them.
Thanks to @limitro!

### CODE REVIEW

File: src/map/clif.cpp [hunk 1]
```diff
@@ -873,14 +873,27 @@ void clif_dropflooritem( struct flooritem_data* fitem, bool canShowEffect ){
...
+
+			for (uint8 i = 0; i < MAX_ITEM_RDM_OPT; i++) {
+				if (fitem->item.option[i].id != 0) {
+					optionCount++;
+				}
+			}
+
+			if (optionCount > 0) {
+				p.showdropeffect = 1;
+				p.dropeffectmode = (optionCount == 1 ? DROPEFFECT_BLUE_PILLAR : optionCount == 2 ? DROPEFFECT_YELLOW_PILLAR : optionCount >= 3 ? DROPEFFECT_PURPLE_PILLAR : DROPEFFECT_NONE) - 1;
```
Comment (secretdataz) [inline src/map/clif.cpp:892]: No offense but this line is 😩

**Review by secretdataz — CHANGES_REQUESTED** [review]

File: src/map/clif.cpp (hunk 1 above)
Comment (aleos89) [reply src/map/clif.cpp:892]: After LGTM complained I figured I’d turn it into a switch 😃

File: src/map/clif.cpp [hunk 2]
```diff
@@ -879,12 +879,32 @@ void clif_dropflooritem( struct flooritem_data* fitem, bool canShowEffect ){
...
+			if (optionCount > 0) {
+				p.showdropeffect = 1;
+				if (optionCount == 1)
+					p.showdropeffect = DROPEFFECT_BLUE_PILLAR - 1;
+				else if (optionCount == 2)
+					p.showdropeffect = DROPEFFECT_YELLOW_PILLAR - 1;
+				else if (optionCount >= 3 && optionCount <= MAX_ITEM_RDM_OPT)
+					p.showdropeffect = DROPEFFECT_PURPLE_PILLAR - 1;
+				else
+					p.showdropeffect = DROPEFFECT_NONE;
```
Comment (Atemo) [inline src/map/clif.cpp:899]:
```suggestion
				else
					p.showdropeffect = DROPEFFECT_PURPLE_PILLAR - 1;
```

File: src/map/clif.cpp [hunk 3]
```diff
@@ -879,12 +879,30 @@ void clif_dropflooritem( struct flooritem_data* fitem, bool canShowEffect ){
...
+			}
+
+			if (optionCount > 0) {
+				p.showdropeffect = 1;
+				if (optionCount == 1)
+					p.showdropeffect = DROPEFFECT_BLUE_PILLAR - 1;
+				else if (optionCount == 2)
+					p.showdropeffect = DROPEFFECT_YELLOW_PILLAR - 1;
+				else
+					p.showdropeffect = DROPEFFECT_PURPLE_PILLAR - 1;
```
Comment (Atemo) [inline src/map/clif.cpp:897]:
```suggestion
					p.dropeffectmode = DROPEFFECT_BLUE_PILLAR - 1;
				else if (optionCount == 2)
					p.dropeffectmode = DROPEFFECT_YELLOW_PILLAR - 1;
				else
					p.dropeffectmode = DROPEFFECT_PURPLE_PILLAR - 1;
```

**Review by Atemo — APPROVED** [review]

**Review by Lemongrass3110 — APPROVED** [review]
Maybe it would be good to have a battle config to enable/disable this?

**Pull request merged**